bench-tps/src/bench.rs

```diff
@@ -402,29 +403,35 @@ fn poll_blockhash<T: Client>(
     client: &Arc<T>,
     id: &Pubkey,
 ) {
-    let mut blockhash_time;
+    let mut blockhash_last_updated = Instant::now();
     loop {
-        blockhash_time = Instant::now();
-        loop {
+        let blockhash_updated = {
             let old_blockhash = *blockhash.read().unwrap();
             if let Ok((new_blockhash, _fee)) = client.get_new_blockhash(&old_blockhash) {
                 *blockhash.write().unwrap() = new_blockhash;
-                break;
+                blockhash_last_updated = Instant::now();
+                true
             } else {
-                if blockhash_time.elapsed().as_secs() > 30 {
-                    panic!("Blockhash is not updating");
+                if blockhash_last_updated.elapsed().as_secs() > 120 {
+                    eprintln!("Blockhash is stuck");
+                    exit(1)
+                } else if blockhash_last_updated.elapsed().as_secs() > 30 {
+                    error!("Blockhash is not updating");
```

Comment:
This will print every 50ms, no?

Reply:
I think it would if the RPC endpoint was totally down. Will fix

Reply:
I think if the endpoint is up, but you still get an old blockhash for say 60 seconds, you will print from 30s->60s every 50ms that "blockhash is not updating". But I think for the user, doing this print every maybe 1-5 seconds would be ideal.

Reply:
Actually `client.get_new_blockhash` will have some timeout, so it will be slower than 50ms, so dunno, maybe it's fine.

Reply:
Better safe than spammy